### ISSUE TITLE
Rule proposal: `prefer-array-find-last`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2245]
### Description

[`Array#findLast`](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast) is available now in Node.js 18.

### Fail

```js
function foo(array, callback) {
  for (let index = array.length - 1; index >= 0; index--) {
    const element = array[index];
    if (callback(element, index, array)) {
      return element;
    }
  }
}
```

```js
let foo;
for (let index = array.length - 1; index >= 0; index--) {
  const element = array[index];
  if (callback(element, index, array)) {
    foo = element;
    break;
  }
}
```

### Pass

```js
array.findLast(callback);
```

### Additional Info

This rule will be difficult to implement.

First, there are too many things in JavaScript that have `.length`, we can't know the type of looping object.

Second, even if we assume they are all arrays, we still can only fix very few cases.

```js
function foo(array, callback) {
  for (let index = array.length - 1; index >= 0; index--) {
    const element = array[index];
    if (callback(element, index, array)) {
      return element;
    }
  }

  return somethingNotUndefiend; // This makes the case unfixable.
}
```

```js
let foo = somethingNotUndefiend; // This makes the case unfixable.
for (let index = array.length - 1; index >= 0; index--) {
  const element = array[index];
  if (callback(element, index, array)) {
    foo = element;
    break;
  }
}
```

**Comment by sindresorhus** [comment]
Accepted

**Comment by sindresorhus** [comment]
As you noted in the issue, this is very difficult to detect reliably. Anything can have `.length`, and most manual loops contain extra logic that makes them unsafe to transform.

Closing as won't-fix as it's not worth the effort.
